Symfony/CS/Tests/Tokenizer/TokensTest.php

```diff
@@ -121,6 +121,78 @@ function (array $matches) {
     }
 
     /**
+     * @dataProvider provideIsShortArrayCases
+     */
+    public function testIsShortArray($source, array $expected)
+    {
+        $tokens = Tokens::fromCode($source);
+
+        foreach ($expected as $index => $expected) {
+            $this->assertSame($expected, $tokens->isShortArray($index));
+        }
+    }
+
+    public function provideIsShortArrayCases()
+    {
+        return array(
+            array(
+                '<?php [];',
+                array(1 => true),
+            ),
+            array(
+                '<?php [1, "foo"];',
+                array(1 => true),
+            ),
+            array(
+                '<?php [[]];',
+                array(1 => true, 2 => true),
+            ),
+            array(
+                '<?php ["foo", ["bar", "baz"]];',
+                array(1 => true, 5 => true),
+            ),
+            array(
+                '<?php (array) [1, 2];',
+                array(3 => true),
+            ),
+            array(
+                '<?php [1,2][$x];',
+                array(1 => true, 6 => false),
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider provideIsNotShortArrayCases
+     */
+    public function testIsNotShortArray($source)
+    {
+        $tokens = Tokens::fromCode($source);
+
+        foreach ($tokens as $index => $token) {
+            if ($token->equals('[')) {
+                $this->assertFalse($tokens->isShortArray($index));
+            }
+        }
```

Comment:
That way it could be easily merged into 2.0 (https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/Symfony/CS/Tests/Tokenizer/Transformer/ArraySquareBraceTransformerTest.php)